mlrun/projects/operations.py

```diff
@@ -210,24 +220,31 @@ def my_pipe(url=""):
             function.spec.command = command
         if local and project and function.spec.build.source:
             workdir = workdir or project.spec.get_code_path()
-        run_result = function.run(
-            name=name,
-            runspec=task,
-            workdir=workdir,
-            verbose=verbose,
-            watch=watch,
-            local=local,
-            artifact_path=artifact_path
-            # workflow artifact_path has precedence over the project artifact_path equivalent to
-            # passing artifact_path to function.run() has precedence over the project.artifact_path and the default one
-            or pipeline_context.workflow_artifact_path
-            or (project.artifact_path if project else None),
-            auto_build=auto_build,
-            schedule=schedule,
-            notifications=notifications,
-            builder_env=builder_env,
-            reset_on_run=reset_on_run,
-        )
+        with warnings.catch_warnings():
+            warnings.simplefilter(
```

Comment:
add a comment that this filter should also be removed when the arguments are deprecated.